app/models/locker_application.rb

```diff
@@ -58,9 +58,10 @@ def available_lockers_in_area_and_size(building:)
     context.order(:location)
   end
 
-  def self.search(uid:)
-    return where(complete: true) if uid.blank?
-
-    where(complete: true).joins(:user).where("users.uid = '#{uid}'")
+  def self.search(uid:, archived:)
+    results = where(complete: true)
+    results = results.where("locker_applications.archived = '#{archived}'") unless ActiveModel::Type::Boolean.new.cast(archived)
```

Comment:
Shouldn't `archived` always already be cast into a boolean at this point? Do we need to re-cast it here?